### PR TITLE
Sync develop changes April 22 - May 7 to hdf5_1_14

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -573,7 +573,6 @@ tools/h5diff/testfiles/tmpSingleSiteBethe.output.h5 -text
 tools/h5diff/testfiles/tmpSingleSiteBethe.reference.h5 -text
 tools/h5diff/testfiles/tmptest.he5 -text
 tools/h5diff/testfiles/tmptest2.he5 -text
-tools/h5dump/errfiles/non_existing.err -text
```
Comment (jhendersonHDF) [inline .gitattributes:576]: Not related to this PR specifically, but shouldn't all these filenames be of the form `tools/test/${tool}/...`? Looks like maybe the paths weren't updated after tools refactoring?

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -1029,10 +1070,67 @@ Bug Fixes since HDF5-1.14.3 release
...
+          ph5diff's output overflow file
+        - an issue where MPI_Init was called after HDF5 had been initialized,
+          preventing the library from setting up an MPI communicator attribute
+          to perform library cleanup on MPI_Finalize
+
     - Renamed h5fuse.sh to h5fuse
 
       Addresses Discussion #3791
 
+    - Fixed an issue with unmatched MPI messages in ph5diff
```
Comment (jhendersonHDF) [inline release_docs/RELEASE.txt]: At least this and the next item below are old fixes in previous releases. I can see the entries in the 1.14.2 and 1.14.3 RELEASE.txt files, but the entries don't appear to be in either the RELEASE.txt or HISTORY-1_14.txt files in the hdf5_1_14 branch.

Comment (lrknox) [reply release_docs/RELEASE.txt]: I see "Fixed an issue with unmatched MPI messages in ph5diff" in the 1.14.3 section and "Fixed an issue in h5repack for variable-length typed datasets" in the 1.14.2 section of HISTORY-1_14.txt in the hdf5_1_14 branch.  I'll remove them - I probably made an incorrect conflict resolution choice.

File: .gitattributes (hunk 1 above)
Comment (byrnHDF) [reply .gitattributes:576]: Looks like that needs fixed in develop!

**Review by byrnHDF — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**